lib/builder/build_plan.go

```diff
@@ -204,17 +156,49 @@ func (plan *BuildPlan) Execute() (*image.DistributionManifest, error) {
 }
 
 func (plan *BuildPlan) executeStage(stage *buildStage, lastStage, copiedFrom bool) error {
+	// Handle `COPY --from=<another_image>`. This case will not work if modifyfs
+	// is set to false, but that combination was rejected in NewPlan().
+	// TODO: This should be done at step level.
+	for alias, _ := range stage.copyFromDirs {
+		if _, ok := plan.stageAliases[alias]; !ok {
+			name, err := image.ParseNameForPull(alias)
```

Comment:
i think stage alias is the name after "as". Here treating `--from=image_name` as an alias can be a little confusing, even though the two are treated similarly. We can probably add more comments for now and think of a nicer way to represent `--from=blah`